table/block_based/block_based_table_reader.h

```diff
@@ -236,6 +236,14 @@ class BlockBasedTable : public TableReader {
       BlockCacheLookupContext* lookup_context, Status s,
       FilePrefetchBuffer* prefetch_buffer) const;
 
+  // input_iter: if it is not null, update this one and return it as Iterator
+  template <typename TBlockIter>
+  TBlockIter* NewDataBlockIterator(
+      const ReadOptions& ro, CachableEntry<Block>& block,
```

Comment:
What if we do not use default values?

Reply:
Yes, we can do that. I'll make them non-optional.